flexget/plugins/filter/series.py

```diff
@@ -1579,13 +1592,36 @@ def auto_exact(self, config):
     def on_task_metainfo(self, task, config):
         config = self.prepare_config(config)
         self.auto_exact(config)
-        for series_item in config:
-            series_name, series_config = list(series_item.items())[0]
-            log.trace('series_name: `%s`, series_config: `%s`', series_name, series_config)
+
+        parser = get_plugin_by_name('parsing').instance
+
+        # Sort Entries into data model similar to https://en.wikipedia.org/wiki/Trie
+        # Only process series if both the entry title and series title first letter match
+        entries_map = {}
+        for entry in task.entries:
+            parsed = parser.parse_series(entry['title'])
+            if parsed.name:
```

Comment:
maybe also check `parsed.valid`, not sure if it'll have a `name` property if it isn't valid